src/core/toga/widgets/optioncontainer.py

```diff
@@ -3,6 +3,63 @@
 from .base import Widget
 
 
+class OptionItem:
+    def __init__(self, container, label, widget, enabled):
+        self._container = container
```

Comment:
The name "container" isn't quite right here. The actual object will be an OptionList, which will be owned by an OptionContainer. 

That said, having a link to the *list* isn't especially helpful - you actually want a reference to the optioncontainer interface.

This variable should also be "None by default", and only assigned when the item is assigned to an optionList. Same goes for `self._index`.